src/base/dev/ne2k/ne2000.c

```diff
@@ -672,7 +672,7 @@ static inline void ne2000_mem_writew(NE2000State *s, uint32_t addr,
 
     addr &= ~1; /* XXX: check exact behaviour if not even */
     if (addr < 32 ||
-        (addr >= NE2000_PMEM_START && addr < NE2000_MEM_SIZE)) {
+        (addr >= NE2000_PMEM_START && addr < (NE2000_MEM_SIZE - 2))) {
```

Comment:
Although in this case it should
then be `<=`?

Reply:
I'm not really sure. Here's what Coverity had to say
![writew](https://github.com/user-attachments/assets/7f41e70f-d806-4fa4-9e13-7f952f985b22)

Reply:
So if you put `<= ... -2` it will
likely not say that.

Reply:
I pushed with `<=` on all four changes.